tests/installation/select_patterns.pm

```diff
@@ -249,35 +210,11 @@ sub select_specific_patterns_by_iteration {
     die "Not all patterns given in the job settings were processed:" . join(", ", @unseen) if @unseen;
 }
 
-sub process_patterns {
-    if (get_var('PATTERNS')) {
-        if (check_var('PATTERNS', 'all') && !check_var('VIDEOMODE', 'text')) {
-            select_all_patterns_by_menu;
-            return 1;
-        }
-        select_specific_patterns_by_iteration;
-    }
-    return 0;
-}
-
-sub process_packages {
-    my $self      = shift;
-    my $secondrun = 0;       # bsc#1029660
-    gotodetails;
-    $secondrun++;
-    $self->gotopatterns;
-    gotodetails;
-    $secondrun--;
-    $self->gotopatterns;
-    gotodetails('unblock', $secondrun);
-}
-
 sub run {
     my ($self) = @_;
-    $self->gotopatterns;
-    return                         if process_patterns;
-    return $self->process_packages if get_var('PACKAGES');
-    accept_changes;
+    $self->go_to_patterns();
+    $self->process_patterns();
```

Comment:
I'm not sure if that's correct. `process_patterns` returns code, but we don't process it here. I haven't check the logic in the method, but if we don't use return code, we can remove it.

Reply:
What I saw is that the return code of process patterns was just to avoid processing packages.
And the return of process packages was to know if changes were made (so that "automatic changes" screen has to be handled.

But, with the split, the return codes loose value.